core/src/data/key.rs

```diff
@@ -50,6 +50,7 @@ pub enum Key {
     Interval(Interval),
     Uuid(u128),
     Inet(IpAddr),
+    Composite(Vec<Key>),
```

Comment:
The idea of adding a `Composite` variant to the `Key` enum is fantastic. I was thinking of a more complex solution, but this approach resolves the problem in a much simpler and more straightforward way. 

The reason `Key` didn't previously support `Value::List` wasn't due to any specific reason; it just wasn't implemented. In this case, how about naming the variant `Key::List` instead of `Key::Composite` to match the `Value` naming? This way, we explicitly support `List` and also handle composite primary keys.

Reply:
Ok!

Reply:
Done